packages/manager-components/src/utils/price.formatter.tsx

```diff
@@ -0,0 +1,182 @@
+import React from 'react';
+import { OsdsText } from '@ovhcloud/ods-components/react';
+import { ODS_TEXT_COLOR_HUE, ODS_TEXT_SIZE } from '@ovhcloud/ods-components';
+import { ODS_THEME_COLOR_INTENT } from '@ovhcloud/ods-common-theming';
+import {
+  OvhSubsidiary,
+  IntervalUnitType,
+  OVH_SUBSIDIARY_CURRENCY,
+} from '../enumTypes';
+import './translations/translations';
+
+export interface PriceOptions {
+  ovhSubsidiary: OvhSubsidiary;
+  locale: string;
+  tFunction: (key: string) => string;
+}
+
+const getPrice = (value: number, tax?: number): number => {
+  const valueWithTax = tax ? value + tax : value;
+  return valueWithTax / 100000000;
+};
+
+const convertIntervalPrice = (
+  price: number,
+  intervalUnit: IntervalUnitType,
+): number => {
+  switch (intervalUnit) {
+    case IntervalUnitType.day:
+      return price / 365;
+    case IntervalUnitType.month:
+      return price / 12;
+    case IntervalUnitType.year:
+      return price;
+    case IntervalUnitType.none:
+      return price;
+    default:
+      return price;
+  }
+};
+
+const getPriceTextFormatted = (
+  ovhSubsidiary: OvhSubsidiary,
+  locale: string,
+  priceValue: number,
+): string => {
+  try {
+    return new Intl.NumberFormat(locale.replace('_', '-'), {
+      style: 'currency',
+      currency: OVH_SUBSIDIARY_CURRENCY[ovhSubsidiary],
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    }).format(priceValue);
+  } catch (e) {
+    const fallbackLocale = 'fr_FR';
+    return new Intl.NumberFormat(fallbackLocale.replace('_', '-'), {
+      style: 'currency',
+      currency: OVH_SUBSIDIARY_CURRENCY[ovhSubsidiary],
+      minimumFractionDigits: 2,
+      maximumFractionDigits: 2,
+    }).format(priceValue);
+  }
+};
+
+const TextPriceContent: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => (
+  <OsdsText
+    size={ODS_TEXT_SIZE._200}
+    color={ODS_THEME_COLOR_INTENT.default}
+    hue={ODS_TEXT_COLOR_HUE._500}
+    className="ml-1"
+  >
+    {children}
+  </OsdsText>
+);
+
+export const formatPrice = (
+  value: number,
+  tax: number,
+  intervalUnit: IntervalUnitType,
+  options: PriceOptions,
+  isConvertIntervalUnit: boolean,
+): string | React.ReactNode => {
+  const { ovhSubsidiary, locale, tFunction } = options;
+
+  const isAsiaFormat = ['ASIA', 'AU', 'IN', 'SG'].includes(ovhSubsidiary);
+  const isGermanFormat = ['DE', 'FI', 'SN'].includes(ovhSubsidiary);
+  const isFrenchFormat = [
+    'CZ',
+    'ES',
+    'FR',
+    'GB',
+    'IE',
+    'IT',
+    'LT',
+    'MA',
+    'NL',
+    'PL',
+    'PT',
+    'TN',
+  ].includes(ovhSubsidiary);
+  const isUSFormat = ['CA', 'QC', 'US', 'WE', 'WS'].includes(ovhSubsidiary);
+
+  let formattedPrice: string | React.ReactNode = '';
+  let convertedValue = value;
+  let convertedTax = tax || 0;
+
+  if (isConvertIntervalUnit) {
+    convertedValue = convertIntervalPrice(value, intervalUnit);
+    convertedTax = convertIntervalPrice(tax || 0, intervalUnit);
+  }
```

Comment:
```suggestion
  const convertedValue = isConvertIntervalUnit ? convertIntervalPrice(value, intervalUnit) : value;
  const convertedTax = isConvertIntervalUnit ? convertIntervalPrice(tax || 0, intervalUnit) : tax || 0;
```

And also you could just take the flag `isConvertIntervalUnit` in the params and add the if directly inside the functions `convertIntervalPrice` and `convertIntervalPrice`